extensions/iidm/src/DiscreteMeasurementsXmlSerializer.cpp

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <powsybl/iidm/extensions/iidm/DiscreteMeasurementsXmlSerializer.hpp>
+
+#include <powsybl/iidm/Enum.hpp>
+#include <powsybl/iidm/Identifiable.hpp>
+#include <powsybl/iidm/converter/Constants.hpp>
+#include <powsybl/iidm/converter/xml/NetworkXmlReaderContext.hpp>
+#include <powsybl/iidm/converter/xml/NetworkXmlWriterContext.hpp>
+#include <powsybl/iidm/extensions/iidm/DiscreteMeasurement.hpp>
+#include <powsybl/iidm/extensions/iidm/DiscreteMeasurementAdder.hpp>
+#include <powsybl/iidm/extensions/iidm/DiscreteMeasurements.hpp>
+#include <powsybl/iidm/extensions/iidm/DiscreteMeasurementsAdder.hpp>
+#include <powsybl/xml/XmlStreamReader.hpp>
+#include <powsybl/xml/XmlStreamWriter.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+namespace extensions {
+
+namespace iidm {
+
+DiscreteMeasurementsXmlSerializer::DiscreteMeasurementsXmlSerializer() :
+    converter::xml::AbstractExtensionXmlSerializer("discreteMeasurements", "network", "dm", "http://www.powsybl.org/schema/iidm/ext/discrete_measurements/1_0") {
+}
+
+Extension& DiscreteMeasurementsXmlSerializer::read(Extendable& extendable, converter::xml::NetworkXmlReaderContext& context) const {
+    extendable.newExtension<DiscreteMeasurementsAdder>().add();
```

Comment:
From my point of view, we should not add the extension before we are sure that the XML is valid. I know that you backport the implementation found in powsybl-core, but I think we have to fix the implementation

Reply:
👍 Agree with you